lab_5_scrapper/scrapper.py

```diff
@@ -204,13 +353,114 @@ def prepare_environment(base_path: Union[pathlib.Path, str]) -> None:
     Args:
         base_path (Union[pathlib.Path, str]): Path where articles stores
     """
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    base_path.mkdir(parents=True)
+
+
+class CrawlerRecursive(Crawler):
+    def __init__(self, config: Config) -> None:
+        """
+        Initialize an instance of the Crawler class.
+
+        Args:
+            config (Config): Configuration
+        """
+        super().__init__(config)
+        self.start_url = self.config.get_seed_urls()[0]
+        self.urls = []
+        self.num_of_urls = 1
+        self.possible_urls = [self.start_url]
+        self.visited_urls = []
+        self.get_info()
+
+    def get_info(self) -> None:
+        if (pathlib.Path(__file__).parent.parent / 'tmp' / 'recursive_crawler.json').exists():
+
+            with open(pathlib.Path(__file__).parent.parent / 'tmp' / 'recursive_crawler.json', 'r', encoding='utf-8') as infile:
+                data = json.load(infile)
+
+            self.urls = data['urls']
+            self.num_of_urls = data['num_visited_urls']
+            self.possible_urls = data['possible_urls']
+            self.visited_urls = data['visited_urls']
+
+    def save_info(self) -> None:
+        data = {
+            'num_visited_urls': self.num_of_urls,
+            'urls': self.urls,
+            'possible_urls': self.possible_urls,
+            'visited_urls': self.visited_urls
+        }
+        with open(pathlib.Path(__file__).parent.parent / 'tmp' / 'recursive_crawler.json', 'w',
```

Comment:
this is the thrid time I see this path. how about DRY?

Reply:
changed it